example/withcouchbase/repository.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"context"
+
+	"github.com/Trendyol/chaki/config"
+	"github.com/Trendyol/chaki/modules/couchbase"
+	"github.com/couchbase/gocb/v2"
+)
+
+type Foo struct {
+	ID  string
+	Bar string
+}
+
+type FooRepository interface {
+	Get(ctx context.Context, id string) (*Foo, error)
+}
+
+type fooRepository struct {
+	coll *gocb.Collection
+}
+
+func NewRepository(cluster *gocb.Cluster, cfg *config.Config) FooRepository {
+	return &fooRepository{
+		coll: cluster.Bucket(cfg.GetString("couchbase.bucketname")).DefaultCollection(),
+	}
+}
+
+func (r *fooRepository) Get(ctx context.Context, id string) (*Foo, error) {
+	result := new(Foo)
+	resp, err := r.coll.Get(id, &gocb.GetOptions{
+		// Enable tracing
+		ParentSpan: couchbase.ParentSpan(ctx),
```

Comment:
We can't turn this usage into something like this `r.coll.Get(ctx, id, &gocb.GetOptions{})` right? Can't we liken it to other usages like `client.Request(ctx)....`

Reply:
We need extra abstraction to convert usage like what we already used.  It means wrapping all gocb.Collection methods. I'm not sure what we need to do here but it is good to have contextual usage as you said